File: backend/src/main/java/de/ropold/backend/dto/AuditLogResponse.java
package de.ropold.backend.dto;

import java.time.LocalDateTime;
import java.util.UUID;

public record AuditLogResponse(
        UUID id,
        String tableName,
        UUID recordId,
        String action,
        String fieldName,
        String oldValue,
        String newValue,
        UserBasicInfo user,
        String username,
        String ipAddress,
        LocalDateTime createdAt
) {
}

File: frontend/src/components/dto/AuditLogResponse.ts
import type {UserBasicInfo} from "./CbcrReportResponse.ts";

export type AuditLogResponse = {
    id: string;
    tableName: string;
    recordId: string;
    action: string;
    fieldName: string;
    oldValue: string;
    newValue: string;
    user: UserBasicInfo;
    username: string;
    ipAddress: string;
    createdAt: string;
}